Manticore11/bjs2 84270 generate user ava

Linempy opened the pull request:
Задача "Интеграция с генератором случайных аватаров юзера"

Linempy committed:
BJS2-84270-generate_user_ava
- Добавил конфигурацию для Minio
- начал реализовывать AvatarService

Linempy committed:
Merge remote-tracking branch 'origin/manticore-master-stream11' into manticore11/BJS2-84270-generate_user_ava

Linempy committed:
BJS2-84270-generate_user_ava
- доделал основную логику

Linempy committed:
BJS2-84270-generate_user_ava
- написал тесты

Linempy committed:
Merge branch 'manticore-master-stream11' into manticore11/BJS2-84270-generate_user_ava

Linempy committed:
BJS2-84270-generate_user_ava
- поправил checkstyle

Linempy committed:
Merge remote-tracking branch 'origin/manticore11/BJS2-84270-generate_user_ava' into manticore11/BJS2-84270-generate_user_ava

RinatBeybutov commented on src/main/java/school/faang/user_service/controller/avatar/AvatarController.java:
```diff
@@ -0,0 +1,47 @@
+     *
+     * @return HTTP 200 OK с данными аватара в теле ответа
+     */
+    @PostMapping
```
тут мы получаем, поэтому лучше get

RinatBeybutov commented on src/main/java/school/faang/user_service/controller/avatar/AvatarController.java:
```diff
@@ -0,0 +1,47 @@
+     */
+    @GetMapping
+    public ResponseEntity<Void> generateAvatar() {
+        service.generateAndSaveAvatar();
```
тут мы создаем, поэтому лучше post

RinatBeybutov commented on src/main/java/school/faang/user_service/service/avatar/AvatarServiceImpl.java:
```diff
@@ -0,0 +1,95 @@
+    private String generateAvatarKey(Long userId) {
+        return String.format(
+                "ava/user_%d_%d.%s",
+                userId, System.currentTimeMillis(), CONTENT_TYPE
```
каждый аргумент с новой строчки, так будет читабельнее

RinatBeybutov commented on src/main/java/school/faang/user_service/service/avatar/DiceBearAvatarService.java:
```diff
@@ -0,0 +1,49 @@
+    private final Random random = new Random();
+
+    private static final List<String> STYLES = List.of(
+            "adventurer", "avataaars", "big-ears", "bottts", "croodles", "micah", "pixel-art"
```
каждая строка с новой строчки

RinatBeybutov commented on src/main/java/school/faang/user_service/service/avatar/DiceBearAvatarService.java:
```diff
@@ -0,0 +1,49 @@
+
+    public DiceBearAvatarService() {
+        this.webClient = WebClient.builder()
+                .baseUrl("https://api.dicebear.com/9.x")
```
этот эндпоинт вынести в переменную класса, а лучше в проперти

RinatBeybutov commented on src/main/java/school/faang/user_service/service/storage/minio/MinioFileStorage.java:
```diff
@@ -0,0 +1,64 @@
+            throws StorageException {
+        try {
+
+            InputStream fileStream = new ByteArrayInputStream(fileBytes);
```
используй try с ресурсами

Linempy committed:
BJS2-84270-generate_user_ava
- вынес ссылку на эндпоинт стороннего сервиса в application.yaml
- поправил замечания

Linempy committed:
BJS2-84270-generate_user_ava
- поправил тесты контроллера
